Editorial review: Document CSS corner-shape (#40663)
mdn/content: files/en-us/web/css/superellipse/index.md

---
title: superellipse()
slug: Web/CSS/superellipse
page-type: css-function
browser-compat: css.types.superellipse
sidebar: cssref
---

> The **`superellipse()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Values_and_Units/CSS_Value_Functions) returns an ellipse shape, which defines the {{cssxref("&lt;corner-shape-value>")}} values used to specify corner shapes.

Author: I tried rewriting this and came up with the following:

> The **`superellipse()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Values_and_Units/CSS_Value_Functions) defines the curvature of an ellipse, and is used to specify [corner shapes](/en-US/docs/Web/CSS/corner-shape) either directly, or via {{cssxref("&lt;corner-shape-value>")}} keywords.

WDYT?

> The **`superellipse()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Values_and_Units/CSS_Value_Functions) returns an ellipse shape, which defines the {{cssxref("&lt;corner-shape-value>")}} values used to specify corner shapes.

Reviewer: Hmm not exactl.

This is the information we should be conveying (which would need some editing)

The superellipse() CSS function defines the curvature of a corner. Its value influences how the corner is shaped:

- A value of 0 creates a straight, diagonal corner (corresponding to the `bevel` keyword).

- Positive values curve the corner outward, making it convex. As the value increases  through `1` (`round`) and `2` (`squircle`) towards `infinity` (`square`), the curvature approaches a perfect right angle.

- Negative values curve the corner inward, making it concave. As the value decreases through `-1` (scoop) towards `-infinity` (`notch`), the curvature sharpens, symmetrical to its positive approach.

## Syntax

```css
superellipse(infinity)
superellipse(4)
superellipse(1.7)
superellipse(0)
superellipse(-2.8)
superellipse(-3)
superellipse(-infinity)
```

### Parameters

- `K`
  - : A number in the range of `-infinity` to `infinity`, inclusive.

### Return value

A superellipse shape.

This is calculated using a modified version of the equation that defines a circle:

x<sup>2</sup> + y<sup>2</sup> = 1

> The circle is made from all points `(x,y)` that satisfy the equation. A given ellipse can be produced by scaling this shape along the x and/or y axis. This is done by replacing the 2 exponent in each case with 2<sup>K</sup>, `K` being the argument passed to `superellipse()`.

Reviewer: Not exactly.

A superellipse is more than an ellipse, as it changes the curvature of the ellipse and rather than scaling it.
See https://en.wikipedia.org/wiki/Superellipse

Author: Ah wait, I've been reading it wrong. I'll update my description now.

---

The **`superellipse()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Values_and_Units/CSS_Value_Functions) defines the curvature of an ellipse, and is used to specify [corner shapes](/en-US/docs/Web/CSS/corner-shape) either directly, or via {{cssxref("&lt;corner-shape-value>")}} keywords.

## Syntax

```css
superellipse(infinity)
superellipse(4)
superellipse(1.7)
superellipse(0)
superellipse(-2.8)
superellipse(-3)
superellipse(-infinity)
```

### Parameters

- `K`
  - : A number in the range of `-infinity` to `infinity`, inclusive.

### Return value

A superellipse shape.

This is calculated using a modified version of the equation that defines a circle:

x<sup>2</sup> + y<sup>2</sup> = 1

A circle is defined by all the points `(x,y)` that satisfy the equation, and a given ellipse can be produced by scaling this shape along the x and/or y axis. A superellipse is created by replacing the 2 exponent in each case with 2<sup>K</sup>, `K` being the argument passed to `superellipse()`, which modifies the curvature of the ellipse.

![Line diagram illustrating the ellipses created using different K values, as described subsequently](superellipse-param.svg)

> - A `K` value of `0` creates a perfect diamond shape, used to create bevelled corners. This corresponds to the {{cssxref("&lt;corner-shape-value>")}} `bevel` keyword.

Author: OK, done

A negative or positive superellipse would appear symmetrical to a superellipse with its inverse value.

## Formal syntax

{{CSSSyntax}}

## Examples

### `superellipse()` value comparison

In this example, we provide two [`<input type="range">`](/en-US/docs/Web/HTML/Reference/Elements/input/range) sliders allowing you to cycle through many different `corner-shape` `superellipse()` values and {{cssxref("border-radius")}} values to compare the effects of each on a container. We've hidden the code for the example here, for brevity, but you can find a [full explanation of it](/en-US/docs/Web/CSS/corner-shape#superellipse_value_comparison) along with other related examples on the {{cssxref("corner-shape")}} reference page.

```html hidden live-sample___value-comparison
<form>
  <div>
    <label for="superellipse-slider">Choose a superellipse() value:</label>
    <input
      type="range"
      id="superellipse-slider"
      min="-5"
      value="0"
      max="5"
      step="0.1" />
  </div>
  <div>
    <label for="radius-slider">Choose a border-radius value:</label>
    <input
      type="range"
      id="radius-slider"
      min="0"
      value="45"
      max="90"

>       step="1" />

Author: I've just removed the step value. I like having it as `1`, because the radius change is nice and smooth. I don't think the max needs to be bigger than it already is.

      <mn>2</mn>
    </msup>
    <mo>+</mo>
    <msup>
      <mi>y</mi>
      <mn>2</mn>
    </msup>
    <mo>=</mo>
    <mn>1</mn>
  </mrow>
</math>

The `x` and `y` variables refer to the `x` and `y` coordinates of points on the circle's circumference; a circle is defined by all the points `(x,y)` that satisfy the equation, and a given ellipse can be produced by scaling this shape along the x and/or y axis. A superellipse is created by replacing the 2 exponent in each case with 2<sup>K</sup>, `K` being the argument passed to `superellipse()`, which modifies the curvature of the ellipse:

<math display="block">
  <mrow>
    <msup>
      <mi>x</mi>
      <msup>
        <mn>2</mn>
        <mi>K</mi>
      </msup>
    </msup>
    <mo>+</mo>
    <msup>
      <mi>y</mi>
      <msup>
        <mn>2</mn>
        <mi>K</mi>
      </msup>
    </msup>
    <mo>=</mo>
    <mn>1</mn>
  </mrow>
</math>

The following diagram illustrates different `superellipse()` values for the top right corner of a container: `infinity`, `1`, `0`, `-1`, and `-infinity`:

![Line diagram illustrating the ellipses created using different K values, as described subsequently](superellipse-param.svg)

> - A `K` value of `0` creates a perfect diamond shape, used to create bevelled corners. This corresponds to the {{cssxref("&lt;corner-shape-value>")}} `bevel` keyword.

Reviewer 2: ```suggestion
- A `K` value of `0` creates a straight line. This value can be used to create bevelled corners and corresponds to the {{cssxref("&lt;corner-shape-value>")}} `bevel` keyword.
```

- A `K` value of `1` creates an ordinary ellipse, corresponding to the `round` keyword.
- A `K` value of `>1` makes the ellipse shape more square; `2` corresponds to the `squircle` keyword.
- A `K` value of `infinity` creates a perfect square (corresponding to the `square` keyword), although `K` values of `10` or more are virtually indistinguishable from a square.
- Negative `K` values result in a concave curve, resulting in corner shapes that are curved inward, or "scooped out". A `K` value of `-1` corresponds to the `scoop` keyword and `-infinity` corresponds to the `notch` keyword.

A negative or positive superellipse would appear symmetrical to a superellipse with its inverse value.

> [!NOTE]
> For any `K` parameter value passed, the `superellipse()` function's return value will always be the same for that `K` value. When that value is applied to two elements, the appearance of the corner shape may differ if the box size or {{cssxref("border-radius")}} values differ.

## Formal syntax

{{CSSSyntax}}

## Examples

### `superellipse()` value comparison

> In this example, two [`<input type="range">`](/en-US/docs/Web/HTML/Reference/Elements/input/range) sliders allow you to cycle through many different `corner-shape` `superellipse()` values and {{cssxref("border-radius")}} values to compare the effects of each on a container. The code is hidden for brevity, but the full explanation of the [superellipse value comparison](/en-US/docs/Web/CSS/corner-shape#superellipse_value_comparison) is provided on the {{cssxref("corner-shape")}} reference page.

Reviewer 2: ```suggestion
In this example, two [`<input type="range">`](/en-US/docs/Web/HTML/Reference/Elements/input/range) sliders allow you to cycle through many different `corner-shape` `superellipse()` values and {{cssxref("border-radius")}} values enabling you to compare the effects of each on a container. The code is hidden for brevity, but the full explanation of the [superellipse value comparison](/en-US/docs/Web/CSS/corner-shape#superellipse_value_comparison) is provided on the {{cssxref("corner-shape")}} reference page.
```
